### PR TITLE
Verify HMAC for OAuth and webhooks

### PR DESCRIPTION
**Pull request by anthonator**

### CODE REVIEW

**Comment by kurt-friedrich** [comment]
Looks great! My only question / suggestion - is `message` the right term for `verify_hmac_for_webhook/3`? It may very well be an idiomatic phrase in this type of authentication (I have no clue), but the Shopify docs only use that term when describing OAuth verification specifically.

In the Shopify example for webhook verification, they use "request contents" and `data` to describe this value. Fwiw I have been using `raw_body` to describe this in my app.

**Comment by anthonator** [comment]
Thanks for the tip! I modified the language to be more consistent with the use-case.

**Comment by kurt-friedrich** [comment]
GitHub is acting up for me, but it looks like the tests still use `message`.

File: test/shopify/rest_test.exs [hunk 1]
```diff
@@ -1,23 +1,33 @@
...
 
     message = "code=0907a61c0c8d55e99db179b68161bc00&shop=some-shop.myshopify.com&state=0.6784241404160823&timestamp=1337178173"
 
     shared_secret = "hush"
 
-    assert { :ok, ^hmac } = Shopify.REST.verify_hmac(hmac, message, shared_secret)
+    assert { :ok, ^hmac } = Shopify.REST.verify_hmac_for_oauth(hmac, message, shared_secret)
+  end
+
+  test "verify_hmac_for_webhook/2" do
```
Comment (kurt-friedrich) [inline test/shopify/rest_test.exs:24]: Typo: `verify_hmac_for_webhook/3`

File: test/shopify/rest_test.exs [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
+    hmac = "ruonad9ilcg3rhfv89nkzi4x7kkh7jibyhxkbewugvi="
+
+    message = "<webhook request body>"
```
Comment (kurt-friedrich) [inline test/shopify/rest_test.exs:27]: message -> body

File: test/shopify/rest_test.exs [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
+
+    shared_secret = "hush"
+
+    assert { :ok, ^hmac } = Shopify.REST.verify_hmac_for_webhook(hmac, message, shared_secret)
```
Comment (kurt-friedrich) [inline test/shopify/rest_test.exs:31]: message -> body

File: README.md [hunk 4]
```diff
@@ -87,36 +87,49 @@ api_key = "a1d5c494473570dde9beb107ebe7d0ba"
...
 
-{:ok, hmac} = Shopify.HMAC.verify(hmac, message, shared_secret)
+{:ok, hmac} = Shopify.REST.verify_hmac_for_oauth(hmac, message, shared_secret)
+```
+
+**`verify_hmac_for_webhook/3`**
+
+```elixir
+hmac = "700e2dadb827fcc8609e9d5ce208b2e9cdaab9df07390d2cbca10d7c328fc4bf"
+message = "<webhook request body>"
```
Comment (kurt-friedrich) [inline README.md:129]: message -> body

File: README.md [hunk 5]
```diff
@@ -87,36 +87,49 @@ api_key = "a1d5c494473570dde9beb107ebe7d0ba"
...
+shared_secret = "hush"
+
+{:ok, hmac} = Shopify.REST.verify_hmac_for_webhook(hmac, message, shared_secret)
```
Comment (kurt-friedrich) [inline README.md:132]: message -> body

**Pull request merged**